src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -183,6 +229,19 @@ private static <T> void enrichDetailsWithErrorDefaults(T defaultValue, FlagEvalu
         details.setReason(Reason.ERROR.toString());
     }
 
+    private static void validateTrackingEventName(String str) {
+        Objects.requireNonNull(str);
+        if (str.isEmpty()) {
+            throw new IllegalArgumentException("trackingEventName cannot be empty");
```

Comment:
https://openfeature.dev/specification/sections/flag-evaluation#requirement-1410 says:

> Methods, functions, or operations on the client MUST NOT throw exceptions, or otherwise abnormally terminate. Flag evaluation calls must always return the default value in the event of abnormal execution. Exceptions include functions or methods for the purposes for configuration or setup.

However, this might be an exception, because:

- it's not a transient error - it's essentially a validation; if this throws it will probably never stop throwing until you fix your code
- it's not an evaluation, and instead a side-effect, and presumably wouldn't break the app totally if it threw, even if you didn't wrap it in a try/catch

Reply:
I agree for exactly the reasons you mentioned. And I've updated the README.md to include this.